pkg/aspect/manager.go

```diff
@@ -28,50 +29,96 @@ import (
 	"istio.io/mixer/pkg/config/descriptor"
 	cpb "istio.io/mixer/pkg/config/proto"
 	"istio.io/mixer/pkg/expr"
-	"istio.io/mixer/pkg/status"
 )
 
 type (
-	// Output captures the output from invoking an aspect.
-	Output struct {
-		// status code
-		Status rpc.Status
-
-		// Additional method-specific returned state
-		Response APIMethodResp
-
-		//TODO attribute mutator
-		//If any attributes should change in the context for the next call
-		//context remains immutable during the call
-	}
-
 	// Manager is responsible for a specific aspect and presents a uniform interface
 	// to the rest of the system.
 	Manager interface {
 		config.AspectValidator
 
-		// NewAspect creates a new aspect instance given configuration.
-		NewAspect(cfg *cpb.Combined, adapter adapter.Builder, env adapter.Env, df descriptor.Finder) (Wrapper, error)
-
-		// Kind return the kind of aspect
+		// Kind return the kind of aspect handled by this manager
 		Kind() Kind
 	}
 
+	// CheckManager take care of aspects used to implement the Check API method
+	CheckManager interface {
+		Manager
+
+		// NewCheckWrapper creates a new aspect wrapper given configuration.
+		NewCheckWrapper(cfg *cpb.Combined, builder adapter.Builder, env adapter.Env, df descriptor.Finder) (CheckWrapper, error)
+	}
+
+	// ReportManager take care of aspects used to implement the Report API method
+	ReportManager interface {
+		Manager
+
+		// NewReportWrapper creates a new aspect wrapper given configuration.
+		NewReportWrapper(cfg *cpb.Combined, builder adapter.Builder, env adapter.Env, df descriptor.Finder) (ReportWrapper, error)
+	}
+
+	// QuotaManager take care of aspects used to implement the Quota API method
+	QuotaManager interface {
+		Manager
+
+		// NewQuotaWrapper creates a new aspect wrapper given configuration.
+		NewQuotaWrapper(cfg *cpb.Combined, builder adapter.Builder, env adapter.Env, df descriptor.Finder) (QuotaWrapper, error)
+	}
+
 	// Wrapper encapsulates a single aspect and allows it to be invoked.
 	Wrapper interface {
 		io.Closer
+	}
 
-		// Execute dispatches to the adapter.
-		Execute(attrs attribute.Bag, mapper expr.Evaluator, ma APIMethodArgs) Output
+	// CheckWrapper encapsulates a single CheckManager aspect and allows it to be invoked.
+	CheckWrapper interface {
```

Comment:
Done, almost. I went with *Executor to avoid my editor claiming a million symbols have typos in them.

All uses of 'wrapper' in this context have been replaced.